modules/dnn/src/layers/convolution_layer.cpp

```diff
@@ -279,9 +292,11 @@ class ConvolutionLayerImpl CV_FINAL : public BaseConvolutionLayerImpl
                      "be multiple of %d but got %d", blobs[0].size[1], inpCn));
         CV_Assert(ngroups > 0 && inpCn % ngroups == 0 && outCn % ngroups == 0);
 
-        int dims[] = {inputs[0][0], outCn, out.height, out.width};
-        outputs.resize(inputs.size(), shape(dims, 4));
-
+        std::vector<int> dims;
+        dims.push_back(inputs[0][0]);
+        dims.push_back(outCn);
+        dims.insert(dims.end(), outShape.begin(), outShape.end());
+        outputs.resize(inputs.size(), shape(&dims[0], dims.size()));
```

Comment:
Let's push `inputs[0][0]` and `outCn` right into `outShape` at the beginning.

Reply:
done